Updates:

birbflock opened the pull request:
1. Refactored configuration settings into a separate file from `user_interface.py` to `config.py`.
2. Implemented functionality to read only the first file from the DICOM series.
3. Enhanced user tips for improved clarity.
4. Added a new function for creating a custom DICOM tag.
5. Created a tutorial for converting streamlit to executable: `tutorial_st2exe.md`.

birbflock committed:
Updates:
1. Refactored configuration settings into a separate file from `user_interface.py` to `config.py`.
2. Implemented functionality to read only the first file from the DICOM series.
3. Enhanced user tips for improved clarity.
4. Added a new function for creating a custom DICOM tag.
5. Created a tutorial for converting streamlit to executable: `tutorial_st2exe.md`.

ml-w commented on dicom_anonymizer/application/anonymizer_utils/anonymize_dicom.py:
```diff
@@ -3,90 +3,108 @@
     # Delete by value group
-    if data_element.VR in va_type:
-        data_element.value = "Annonymized"
+    if data_element.VR.strip() in [v.strip() for v in va_type]:
```
Better keep it as "Anonymized", for fields that are not string, you may set it as empty. 
You can do it with a `try-except` loop

ml-w commented on dicom_anonymizer/application/anonymizer_utils/anonymize_dicom.py:
```diff
@@ -124,38 +142,14 @@ def anonymize(file_dir, output_dir, tags=None, update: Optional[dict] = None, ta
-        >>> annonymize(folder, out_folder, tags, update=update)
-
     """
+    
```
Delete

ml-w commented on dicom_anonymizer/application/config.py:
```diff
@@ -0,0 +1,39 @@
+    'PatientID':        '',                                     # for user's inputs
+    'InstitutionName':  '',                                     # for user's inputs
+    'PatientBirthDate': '19700101',                             # reset patient's birth date to 0
+    'AccessionNumber':  lambda x: x[3:]                         # remove the first 3 characters        
```
Some hospital has less than 3 characters, so this is easily broken. Try using regular expression like `^[a-zA-Z]+` and `re.sub` for this task

ml-w commented on dicom_anonymizer/application/ui_utils/ui_logic.py:
```diff
@@ -1,14 +1,32 @@
+            
 
-def update_data_editor(edit_df, upload_df, update_tags):
+def update_data_editor(edit_df: pd.DataFrame, upload_df: pd.DataFrame, update_tags: dict):
```
Good habit to mark the datatype, we also like to use -> XXX to show the returned datatype.

ml-w commented on dicom_anonymizer/application/ui_utils/ui_logic.py:
```diff
@@ -21,23 +39,79 @@ def update_data_editor(edit_df, upload_df, update_tags):
+    return empty_col
+
+def validate_upload(edit_df, upload_df, update_tags, upload_df_id):
+    """
```
Generally, there are a few docstring style. For this project, try to stick to Google's style and Napolean Sphinx syntax.

ml-w commented on dicom_anonymizer/application/user_interface.py:
```diff
@@ -1,9 +1,10 @@
+
 from anonymizer_utils.anonymize_dicom import *
 from ui_utils.ui_logic import *
+from config import unique_ids, ref_tags, update_tags, upload_df_id, tags_2_anon, tags_2_spare, new_tags
```
If it's package specific, general practice is to attach it with a '.' so that the import would not be overriden by default packages. e.g., `import .config` 

Ensure that config.py or ./config is existing file/package folder

birbflock committed:
Updates (amended based on comments):
1. Created configuration settings: `app_settings.config`.
2. Shortened file reading process time: read only the first file from the DICOM series.
3. Improved user tips with step-by-step guide.
4. Added a new function: create a new DICOM tag.
5. Created a tutorial for converting streamlit to executable: `tutorial_st2exe.md`.

birbflock committed:
Amend a mistake in config.py
